File: experiments/llm-reasoning-cost/generated_code/python/06_struct_methods_codex.py
from dataclasses import dataclass
from typing import List


@dataclass
class Point:
    x: int
    y: int

    def squared_mag(self) -> int:
        return self.x * self.x + self.y * self.y


def filter_and_sum(points: List[Point], max_mag: int) -> int:
    total = 0
    for p in points:
        mag = p.squared_mag()
        if mag <= max_mag:
            total += mag
    return total

File: experiments/llm-reasoning-cost/generated_code/cpp/06_struct_methods_codex.cpp
#include <cstdint>
#include <vector>

struct Point {
    int32_t x;
    int32_t y;

    int32_t squared_mag() const {
        return x * x + y * y;
    }
};

int32_t filter_and_sum(const std::vector<Point>& points, int32_t max_mag) {
    int32_t total = 0;
    for (const auto& p : points) {
        int32_t mag = p.squared_mag();
        if (mag <= max_mag) {
            total += mag;
        }
    }
    return total;
}